python/semantic_kernel/connectors/ai/open_ai/open_ai_response.py

```diff
@@ -0,0 +1,210 @@
+        return self.raw_response.choices[0].text
+
+    @property
+    def all_content(self) -> List[Optional[str]]:
+        """Get the content of the response."""
+        if not isinstance(self.raw_response, Completion):
+            raise ValueError("all_content is not available for streaming responses, use stream_content instead.")
+        return [choice.text for choice in self.raw_response.choices]
+
+    async def streaming_content(self) -> AsyncGenerator[str, None]:
+        """Get the streaming content of the response."""
+        if isinstance(self.raw_response, Completion):
+            raise ValueError("streaming_content is not available for regular responses, use content instead.")
+        if self.stream_content is None:
+            self.stream_content = ""
+        async for chunk in self.raw_response:
+            if len(chunk.choices) == 0:
+                continue
+            if chunk.choices[0].text:
+                text = chunk.choices[0].text
+                if text.strip():
+                    self.stream_content += text
+                    yield text
+
+    async def streaming_all_content(self) -> AsyncGenerator[List[str], None]:
+        """Get the streaming content of the response."""
+        if isinstance(self.raw_response, Completion):
+            raise ValueError("streaming_all_content is not available for regular responses, use all_content instead.")
+        if self.stream_all_content is None:
+            self.stream_all_content = [""] * self.request_settings.number_of_responses
+        async for chunk in self.raw_response:
+            if len(chunk.choices) == 0:
+                continue
+            current_chunks: List[str] = []
+            for choice in chunk.choices:
+                if choice.text:
+                    text = choice.text
+                    if text.strip():
+                        self.stream_all_content[choice.index] += text
+                        current_chunks.append(text)
+            yield current_chunks
+
+
+class OpenAIChatResponse(AIResponse):
+    """A response from OpenAI."""
+
+    raw_response: Union[ChatCompletion, AsyncStream[ChatCompletionChunk]]
+    stream_content: Optional[str] = None
+    stream_all_content: Optional[List[str]] = None
+
+    @property
+    def content(self) -> Optional[str]:
+        """Get the content of the response."""
+        if not isinstance(self.raw_response, ChatCompletion):
+            raise ValueError("content is not available for streaming responses, use stream_content instead.")
+        return self.raw_response.choices[0].message.content
+
+    @property
+    def function_call(self) -> Optional[FunctionCall]:
+        """Get the function call of the response."""
+        if not isinstance(self.raw_response, ChatCompletion):
+            raise ValueError(
+                "function_call is not available for streaming responses, use stream_function_call instead."
```

Comment:
Is the `stream_function_call` a to-do item. I don't see it here?

Reply:
indeed, still rough!